### PR TITLE
Use OptionParser for command line option handling

### PR DESCRIPTION
**Pull request by Mopster**

### CODE REVIEW

File: vnet/bin/vnflows-monitor [hunk 1]
```diff
@@ -464,57 +442,30 @@ def output_flows(outputflows)
...
 def do_sleep
-  sleep $opts.interval
+  sleep $opts[:time]
 end
 
 def update_options
   while $stdin.ready?
-    $opts.parse_options $stdin.readline
+    $stdin.readline
   end
```
Comment (triggers) [inline vnet/bin/vnflows-monitor]: The idea here was to make the tool more interactive.  For example, you could type "t 1 <ret>" while the program was running and just change the time delay without restarting.  If there were various levels of pretty printing implemented you could change them on-the-fly while debugging.  Maybe there is no easy way to keep this feature with OpenParser, in which case you should just delete the update_options method.

File: vnet/bin/vnflows-monitor [hunk 2]
```diff
@@ -464,57 +442,30 @@ def output_flows(outputflows)
...
-$opts.parse_options(ARGV.to_s)
-
-iters = $opts.iterationselect
-if iters.length != 0
-  if $opts.runselect == 0
+if $opts[:iteration]
+  put "iterations"
+  iters = $opts[:iteration]
+  if $opts[:run] == 0
     latest = latest_run_number
```
Comment (triggers) [inline vnet/bin/vnflows-monitor]: To use the diff feature on archived data, you need to specify two iterations.  That is why $opts.iterationselect was an array.  See the example in the help "d 1 i xxx yyy r zzz".  Both xxx and yyy would get appended to the array.  If you want to break this feature (as the current pull request does), the code in the "iters.length" below can be removed.

File: vnet/bin/vnflows-monitor [hunk 3]
```diff
@@ -394,7 +344,35 @@ def do_diff(previous, current)
...
-$opts = VnflowsMonitorOptions.new
+$opts = {}
+$opts[:count] = 1
+$opts[:head] = 0
+$opts[:pretty] = 1
+$opts[:time] = 2
+$opts[:run] = 0
+OptionParser.new do |op|
+  op.on("-c", "--count N", Float, "Number of iterations, 0 = keep going (default 1") do |c|
+    $opts[:count] = c
```
Comment (triggers) [inline vnet/bin/vnflows-monitor]: Missing )

File: vnet/bin/vnflows-monitor (hunk 2 above)
Comment (Mopster) [inline vnet/bin/vnflows-monitor]: Changed the command line to accept an array input

File: vnet/bin/vnflows-monitor [hunk 4]
```diff
@@ -394,7 +344,35 @@ def do_diff(previous, current)
...
+    $opts[:pretty] = p
+  end
+  op.on("-t", "--time N", Float, "Time in seconds to pause between iterations (default 2)") do |t|
+    $opts[:time] = t
+  end
+  op.on("-r", "--run S", "Pull flows from previous run number xxx, defaults to most recent run") do |r|
+    $opts[:run] = r
+  end
+  op.on("-i", "--iteration x,y,z", Array, "Pull flows from iteration number x. y and z") do |i|
+    $opts[:iteration] = i
```
Comment (triggers) [inline vnet/bin/vnflows-monitor]: Either "-i x" which implies -i (x-1), x  OR "-i x,y".   If more than two are given, only the last two are used.

The old code used only Integers.  I assume you are testing this with the floats and strings.

**Comment by akry** [comment]
I think we should avoid using global variables in order not to collide with system's global variables.

https://github.com/bbatsov/ruby-style-guide#instance-vars

**Comment by akry** [comment]
+1

**Comment by Metallion** [comment]
I'm going to quickly test this before merging, just to have used the new commandline options once. It's been unmerged long enough so I'll make sure to do this today.

**Comment by Metallion** [comment]
I should have done this before asking for a review but this happened.

```
./vnflows-monitor 
./vnflows-monitor:353:in `attr_accessor': invalid attribute name `$un_n' (NameError)
        from ./vnflows-monitor:353:in `singleton class'
        from ./vnflows-monitor:347:in `<module:Options>'
        from ./vnflows-monitor:346:in `<main>'
```

Let's see if I can quickly fix that.

**Comment by Metallion** [comment]
rpmbuild failed when trying to upload the packages. Rerunning.

**Pull request merged**